docs/lang/static.md

```diff
@@ -1,16 +1,84 @@
+This is in contrast to a *latency-sensitive*, or *static*, model of computation, where the number of cycles a component needs is known to, and honored by, the compiler.
 
-More crucially, however, it is impossible for *latency-insensitive* programs to interact with *latency-sensitive* hardware implemented in RTL.
+In general, latency-insensitivity makes it easier to compose programs.
+It grants the compiler freedom to schedule operators however it wants, as long as it meets the program's dataflow constraints.
 
-Calyx uses the `@static` attribute to provide latency information to various constructs and provides strong guarantees about the generated programs.
+However, there are two drawbacks to this approach.
+First, the generated hardware may not be efficient: if the compiler does not know how long computations take, it must schedule them conservatively.
+Second, it is impossible for *latency-insensitive* programs to interact with *latency-sensitive* hardware implemented in RTL;
+this prevents the use of black-box hardware designs.
+
+To address these issues, Calyx provides a `static` keyword that modifies components and groups, along with static variants of other control operators.
+
+## Static support in the Calyx IL
+
+### Static components
+
+Say we have a multiplier component, `std_mult`, which multiplies the values `left` and `right` and puts the result in `out`.
+Its latency is 3 cycles.
+We can declare it as follows:
+```
+static<3> primitive std_mult[W](go: 1, left: W, right: W) -> (out: W);
+```
+Compare this to the divider component `std_div`, whose latency is unknown:
+```
+primitive std_div[W](go: 1, left: W, right: W) -> (out: W, done: 1)
+```
+The key differences are:
+- The `static` keyword is used to declare the component as static and to specify its latency.
+- The `done` port is not present in the static component.
+
+A client of the divider must pass two inputs, raise the `go` signal, and wait for the component itself to raise its `done` signal.
+In contrast, a client of the multiplier must pass two inputs and raise the `go` signal, but it does not need to wait for the component to raise a `done` signal.
+It can simply and safely assume that the result will be available after 3 cycles.
+
+
+### Static groups and relative timing guards
+
+Much like components, groups can be declared as static.
+Since groups are just unordered sets of assignments, it pays to have a little more control over the scheduling of the assignments within a group.
+Consider this group, which performs `ans := 6 * 7`:
+```
+static<4> group mult_and_store {
+  mult.left = %[0:3] ? 6;
+  mult.right = %[0:3] ? 7;
+  mult.go = %[0:3] ? 1;
+  ans.in = %3 ? mult.out;
+  ans.write_en = %3 ? 1;
+}
+```
+The `static<4>` keyword specifies that the group should take 4 cycles to execute.
+However, we cannot just let all the assignments execute in any order, as we could with a dynamic group.
```

Comment:
I'd probably delete this sentence… I'm not sure "let all the assignments execute in any order" is truly a differentiating quality between static & dynamic. The literal order of assignments is irrelevant in both settings, for instance.